Assets/OngSharing/Singleton.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Be aware this will not prevent a non singleton constructor
+///   such as `T myT = new T();`
+/// To prevent that, add `protected T () {}` to your singleton class.
+/// 
+/// As a note, this is made as MonoBehaviour because we need Coroutines.
+/// </summary>
+public class Singleton<T> : MonoBehaviour where T : MonoBehaviour
```

Comment:
@mr0ng let's also get rid of this and replace whatever uses it with a `IMixedRealityNetworkingSystem`